### ISSUE TITLE
no-useless-escape false positive on closing bracket

### ISSUE THREAD

**Issue by dbolton** [eslint/eslint#12340]
**Tell us about your environment**

* **ESLint Version:**
* **Node Version:** v10.16.3
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?** linter-js-standard for Atom also the command line npx standard

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
var test = /[\][]/
```

**What did you expect to happen?**
No errors

**What actually happened? Please include the actual, raw output from ESLint.**
The error: 
```
Unnecessary escape character: \].
```
If I remove the escape character, I get a very different result in JavaScript. For example, 

```js
'x][x'.replace(/[\][]+/, '') === 'xx'
'x][x'.replace(/[][]+/, '') === 'x][x'
```

**Are you willing to submit a pull request to fix this bug?**
No

**Related bugs**

- https://github.com/eslint/eslint/issues/11627 is a similar report about the opening square bracket. Technically you can get away with escaping the opening square bracket, but not with escaping the closing square bracket as reported in the current issue. 
- Downstream report in StandardJS: https://github.com/standard/standard/issues/1426

**Comment by platinumazure** [comment]
Hi @dbolton, thanks for the issue.

I [can't reproduce](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVzZWxlc3MtZXNjYXBlOiBbXCJlcnJvclwiXSovXG5cbnZhciB0ZXN0ID0gL1tcXF1bXS8iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEwLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19) in the online demo.

Could you please let me know if I'm missing a key piece of configuration, and/or share more details about your configuration to help narrow down the problem? Thanks!

**Comment by dbolton** [comment]
Now I can't reproduce either.
